### PR TITLE
Clean up and add links for compartmental model docs

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR fixes a few formatting issues with the model documentation and adds a link on Read the Docs for the PyNEST examples.

### CODE REVIEW

File: models/cm_default.h [hunk 1]
```diff
@@ -90,36 +94,43 @@ Synaptic receptors can be added as follows
...
-  that specifies the type of receptor. In the `nest.Connect()` call, the
-  'receptor_type' entry is and integer that specifies the receptor index.
+.. note::
 
-.. note:: The "compartments" resp. "receptors" entries can be a dict or a list
-  of dicts containing compartment resp. receptor details. When a dict is provided,
-  a single compartment resp. receptor is added to the model. When a list of dicts
-  is provided, multiple compartments resp. receptors are added with a single
-  `nest.SetStatus()` call.
+  In the ``nest.SetStatus()`` call, the ``receptor_type`` entry is string
```
Comment (clinssen) [inline models/cm_default.h:107]:
```suggestion
  In the ``nest.SetStatus()`` call, the ``receptor_type`` entry is a string
```

File: models/cm_default.h [hunk 2]
```diff
@@ -90,36 +94,43 @@ Synaptic receptors can be added as follows
...
+  that specifies the type of receptor. In the ``nest.Connect()`` call, the
+  ``receptor_type`` entry is and integer that specifies the receptor index.
```
Comment (clinssen) [inline models/cm_default.h:109]:
```suggestion
  ``receptor_type`` entry is an integer that specifies the receptor index.
```

File: models/cm_default.h [hunk 3]
```diff
@@ -90,36 +94,43 @@ Synaptic receptors can be added as follows
...
 
 Compartment voltages can be recorded. To do so, create a multimeter in the
-standard manner but specify the to be recorded voltages as
-'v_comp{compartment_index}'. Ion channels state variables can be recorded as well,
-using the syntax '{state_variable_name}{compartment_index}'. For receptor state
-variables, use the receptor index '{state_variable_name}{receptor_index}' i.e.
+standard manner but specify the recorded voltages as
+``v_comp{compartment_index}``. Ion channels state variables can be recorded as well,
+using the syntax ``{state_variable_name}{compartment_index}``. For receptor state
+variables, use the receptor index ``{state_variable_name}{receptor_index}`:`
```
Comment (clinssen) [inline models/cm_default.h:123]:
```suggestion
variables, use the receptor index ``{state_variable_name}{receptor_index}``:
```

**Review by clinssen — CHANGES_REQUESTED** [review]

File: models/cm_default.h [hunk 4]
```diff
@@ -90,36 +94,43 @@ Synaptic receptors can be added as follows
...
+  In the ``nest.SetStatus()`` call, the ``receptor_type`` entry is a string
+  that specifies the type of receptor. In the ``nest.Connect()`` call, the
+  ``receptor_type`` entry is an integer that specifies the receptor index.
+
+.. note::
+
+  Each compartments' respective "receptors" entries can be a dictionary or a list
+  of dictionaries containing receptor details. When a dictionary is provided,
+  a single compartment receptor is added to the model. When a list of dicts
+  is provided, multiple compartments receptors are added with a single
```
Comment (jougs) [inline models/cm_default.h:116]:
```suggestion
  is provided, multiple compartments' receptors are added with a single
```

File: models/cm_default.h [hunk 5]
```diff
@@ -90,36 +94,43 @@ Synaptic receptors can be added as follows
...
+  ``nest.SetStatus()`` call.
 
 Compartment voltages can be recorded. To do so, create a multimeter in the
-standard manner but specify the to be recorded voltages as
-'v_comp{compartment_index}'. Ion channels state variables can be recorded as well,
-using the syntax '{state_variable_name}{compartment_index}'. For receptor state
-variables, use the receptor index '{state_variable_name}{receptor_index}' i.e.
+standard manner but specify the recorded voltages as
+``v_comp{compartment_index}``. Ion channels state variables can be recorded as well,
```
Comment (jougs) [inline models/cm_default.h:121]:
```suggestion
``v_comp{compartment_index}``. State variables for ion channels can be recorded as well,
```

File: models/cm_default.h [hunk 6]
```diff
@@ -132,7 +143,7 @@ The following parameters can be set in the status dictionary.
  V_th       mV      Spike threshold (default: -55.0 mV)
 =========== ======= ===========================================================
 
-The following parameters can be set using the `AddCompartment` function
+The following parameters can be set using the ``AddCompartment`` function
```
Comment (jougs) [inline models/cm_default.h:146]: There is no ``AddCompartment`` function anymore. This is now done via the call to ``SetStatus()``.
```suggestion
The following parameters can be used when adding compartments using ``SetStatus()``
```

**Review by jougs — COMMENTED** [review]
Looks mostly good to me.

**Review by jougs — APPROVED** [review]
LGTM!

**Review by clinssen — APPROVED** [review]

**Pull request merged**